### PR TITLE
Tmp draft PR for Nerdbank.GitVersioning question

### PR DESCRIPTION
**Pull request by MarcoRossignoli**
Tmp PR to understand issue with Nerdbank.GitVersioning for nighly build versioning.

### CODE REVIEW

File: src/coverlet.collector/version.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+    "$schema": "https://raw.githubusercontent.com/AArnott/Nerdbank.GitVersioning/master/src/NerdBank.GitVersioning/version.schema.json",
+    "version": "1.0.0",
```
Comment (onovotny) [inline src/coverlet.collector/version.json]: This is using a fixed version of 1.0.0. If you want NB.GV to work, you should specify either `1.0` or `1.0.0-build.{height}`

**Comment by onovotny** [comment]
By specifying the three part version, NB.GV won't do anything automatically. It increments the patch version and you've specified it. If you specify only the Major.Minor, NB.GV will set the patch version.

File: src/coverlet.collector/version.json (hunk 1 above)
Comment (MarcoRossignoli) [reply src/coverlet.collector/version.json]: Did it, but no luck 
image

File: src/coverlet.collector/version.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+    "$schema": "https://raw.githubusercontent.com/AArnott/Nerdbank.GitVersioning/master/src/NerdBank.GitVersioning/version.schema.json",
+    "version": "1.0.0-build.{height}",
```
Comment (MarcoRossignoli) [inline src/coverlet.collector/version.json]: @onovotny did it but no luck
image

**Comment by AArnott** [comment]
I don't see your nuspec file in this PR. How do you set its version tag? It should be set to `$version$` I believe.

**Comment by AArnott** [comment]
Ya, here is your problem :

https://github.com/tonerdo/coverlet/blob/0271f5eef162788b56fb7707b0475c7288fd8d62/src/coverlet.collector/coverlet.collector.nuspec#L5

**Comment by MarcoRossignoli** [comment]
@AArnott no lock with `$version$`
image
I think `pack` command expect version param as command input...and `$version$` is only a placeholder https://docs.microsoft.com/en-us/nuget/reference/nuspec#replacement-tokens

**Comment by AArnott** [comment]
You're closer. Now per [this doc](https://docs.microsoft.com/en-us/nuget/reference/msbuild-targets#packing-using-a-nuspec) you should set a project property like this:

```xml
<PropertyGroup>
  <NuspecProperties>$(NuspecProperties);version=$(NuGetPackageVersion)</NuspecProperties>
</PropertyGroup>
```

**Comment by MarcoRossignoli** [comment]
@AArnott closed but not enough seem that prop is empty 
image
image

**Comment by AArnott** [comment]
Ah, ok. Try adding `InitialTargets="GetBuildVersion"` to your `<Project>` tag at the top of the file.

**Comment by AArnott** [comment]
Oh, that won't work. Hang on... Do this:

```xml
<Target Name="SetNuspecPackageVersion" DependsOnTargets="GetBuildVersion" BeforeTargets="Pack">
  <PropertyGroup>
    <NuspecProperties>$(NuspecProperties);version=$(NuGetPackageVersion)</NuspecProperties>
  </PropertyGroup>
</Target>
```

**Comment by MarcoRossignoli** [comment]
it does no trigger target maybe because `GetBuildVersion` run inside `Pack`
image

**Comment by AArnott** [comment]
How are you building this? 

The build authoring in this repo looks needlessly complex. Why does build.proj manually run all these Exec tasks rather than simply invoking the targets on each project? As it is, the same targets get repeatedly built at each of a few steps. 

When I use `dotnet pack` (which *ought* to be what's used here, probably), all the versions are perfect. If coverlet.collector needs its own nuspec file, this should be specified in the project file itself, not the build.proj as a special parameter.

Anyway, [here's the minimum](https://github.com/AArnott/coverlet/commit/85e5f3445807bea1d78f72b09a79ab6c5267eb23) I had to add to get the package version for coverlet.collector correct. I had to make one change to what I had prescribed earlier.

**Comment by AArnott** [comment]
You might want to take a few more commits from [my nightly branch](https://github.com/AArnott/coverlet/commits/nightly). Which is based on yours.

**Comment by MarcoRossignoli** [comment]
> The build authoring in this repo looks needlessly complex. Why does build.proj manually run all these Exec tasks rather than simply invoking the targets on each project? As it is, the same targets get repeatedly built at each of a few steps.

I agree, we're working on this direction.

> When I use dotnet pack (which ought to be what's used here, probably), all the versions are perfect. If coverlet.collector needs its own nuspec file, this should be specified in the project file itself, not the build.proj as a special parameter.

I agree will do

> You might want to take a few more commits from my nightly branch. Which is based on yours.

Sure thank you very much for your fundamental help and for your great tool!

**Comment by AArnott** [comment]
You're welcome. I'm actually preparing a few more commits to clean up the build authoring. Are you interested, or would you rather take over here?

**Comment by MarcoRossignoli** [comment]
Feel free to commit your idea in you repo and we'll pick up, maybe not immediatly, but sure in next PR also because we will deploy new visual studio agnostic build version https://github.com/tonerdo/coverlet/pull/396 and we'll need to fix some path.
We need to deploy ASAP a nighly build because we will release VSTest official cross coverage collector 
https://github.com/tonerdo/coverlet/issues/395 
https://github.com/tonerdo/coverlet/issues/395#issuecomment-495602245
and we would have a way to provide nighly build to fix possible new bugs, we expect some "load" if MS will advertise this new `Xplat Code Coverage` and we want to be "ready" to provide best experience we can.

One more question on versioning...I would like to have coherent package name but for collector I get different output 
image
But version.json are the same.

**Comment by MarcoRossignoli** [comment]
> Are you interested, 

We are always interested in valuable enhancement FYI here you can take a look on new docs with great explanation of coverlet features https://github.com/tonerdo/coverlet/pull/422 by @tonerdo

**Comment by MarcoRossignoli** [comment]
It works!Close this draft I'll re-open official clean PR thank's @AArnott @onovotny

**Pull request closed without merging**